adagrams/game.py

```diff
@@ -1,11 +1,174 @@
+import random
+
+LETTER_POOL_W_VALUES = {
+    'A': {
+        'qty': 9, 
+        'value': 1
+    },
+    'B': {
+        'qty': 2,
+        'value': 3
+    },
+    'C': {
+        'qty': 2, 
+        'value': 3
+    },
+    'D': {
+        'qty': 4, 
+        'value': 2
+    },
+    'E': {
+        'qty': 12,
+        'value': 1
+    },
+    'F': {
+        'qty': 2, 
+        'value': 4
+    },
+    'G': {
+        'qty': 3, 
+        'value': 2
+    },
+    'H': {
+        'qty': 2, 
+        'value': 4
+    },
+    'I': {
+        'qty': 9,
+        'value' :1
+    },
+    'J': {
+        'qty': 1, 
+        'value': 8
+    },
+    'K': {
+        'qty': 1, 
+        'value': 5
+    },
+    'L': {
+        'qty': 4, 
+        'value': 1
+    },
+    'M': {
+        'qty': 2, 
+        'value': 3
+    },
+    'N': {
+        'qty': 6, 
+        'value': 1
+    },
+    'O': {
+        'qty': 8,
+        'value': 1
+    },
+    'P': {
+        'qty': 2, 
+        'value': 3
+    },
+    'Q': {
+        'qty': 1, 
+        'value': 10
+    },
+    'R': {
+        'qty': 6,
+        'value': 1
+    },
+    'S': {
+        'qty': 4, 
+        'value': 1
+    },
+    'T': {
+        'qty': 6, 
+        'value': 1
+    },
+    'U': {
+        'qty': 4, 
+        'value': 1
+    },
+    'V': {
+        'qty': 2, 
+        'value': 4
+    },
+    'W': {
+        'qty': 2, 
+        'value': 4
+    },
+    'X': {
+        'qty': 1, 
+        'value': 8
+    },
+    'Y': {
+        'qty': 2, 
+        'value': 4
+    },
+    'Z': {
+        'qty': 1,
+        'value': 10
+    }
+}
+
 def draw_letters():
-    pass
+    letters = []
+    frequency_of_letters = []
+    for letter, frequency in LETTER_POOL_W_VALUES.items():
+        letters.append(letter)
+        frequency_of_letters.append(frequency['qty'])
+    # random.sample returns a new list containing elements from the population
+    # while leaving the original population unchanged. counts controls the
+    # probability of an element within the sample. in this case, the greater
+    # the quantity of tiles for a letter, the higher probability that letter
+    # will be part of the random sample.
+    random_letters_list = random.sample(letters, counts =\
+        frequency_of_letters, k = 10)
+    return random_letters_list
 
 def uses_available_letters(word, letter_bank):
-    pass
+    copy_letter_bank = letter_bank.copy()
+    for letter in word.upper():
+        if letter not in copy_letter_bank:
+            return False
+        else: 
+            copy_letter_bank.remove(letter)
+    return True
 
 def score_word(word):
-    pass
+    score = 0
+    for letter in word.upper():
+        for k, v in LETTER_POOL_W_VALUES.items():
```

Comment:
This loop works for sure, but you could save the additional complexity of the inner for-loop and just use the dictionary itself, e.g.
```python
        score += LETTER_POOL_W_VALUES[letter]["value"]
```